src/nools/lib.js

```diff
@@ -73,15 +73,15 @@ function emitTasksForSchedule(c, schedule, r) {
 
       if(r) {
         if(event.dueDate) {
-          dueDate = event.dueDate(r, event, scheduledTaskIdx);
+          dueDate = event.dueDate(c, r, event, scheduledTaskIdx);
```

Comment:
It seems confusing to me that `dueDate` would receive a different collection of parameters, depending on the situation. 
Also, adding an extra parameter for `reports` also appears to break existing configurations?

Reply:
Ya.. I agree. Updating to move all optional parameters to the end of the interface.